proposed/http-factory/http-factory.md

```diff
@@ -105,48 +116,44 @@ Has the ability to create streams for requests and responses.
 ```php
 namespace Psr\Http\Message;
 
+use Psr\Http\Message\StreamInterface;
+
 interface StreamFactoryInterface
 {
     /**
-     * Create a new stream with no content.
-     *
-     * The stream will be writable and seekable.
-     *
-     * @return StreamInterface
-     */
-    public function createStream();
-
-    /**
-     * Create a new stream from a callback.
+     * Create a new stream from a string.
      *
-     * The stream will be read-only and not seekable.
+     * The stream SHOULD be created with a temporary resource.
      *
-     * @param callable $callback
+     * @param string $content
      *
      * @return StreamInterface
      */
-    public function createStreamFromCallback(callable $callback);
+    public function createStream($content = '');
 
     /**
-     * Create a new stream from a resource.
+     * Create a stream from an existing file.
      *
-     * @param resource $body
+     * The file MUST be opened using the given mode, which may be any mode
+     * supported by the `fopen` function.
+     *
+     * @param string $file
```

Comment:
Should the `$file` string be interpreted the same way as in the `fopen()` function? http://php.net/manual/en/function.fopen.php#refsect1-function.fopen-parameters